poll/poll.py

```diff
@@ -0,0 +1,849 @@
+                for key, value in items]
+
+    @staticmethod
+    def gather_items(data, result, noun, field, image=True):
+        """
+        Gathers a set of label-img pairs from a data dict and puts them in order.
+        """
+        items = []
+        if field not in data or not isinstance(data[field], list):
+            source_items = []
+            result['success'] = False
+            result['errors'].append(
+                "'{0}' is not present, or not a JSON array.".format(field))
+        else:
+            source_items = data[field]
+
+        # Make sure all components are present and clean them.
+        for item in source_items:
+            if not isinstance(item, dict):
+                result['success'] = False
+                result['errors'].append(
+                    "{0} {1} not a javascript object!".format(noun, item))
+                continue
+            key = item.get('key', '').strip()
+            if not key:
+                result['success'] = False
+                result['errors'].append(
+                    "{0} {1} contains no key.".format(noun, item))
+            image_link = item.get('img', '').strip()
+            label = item.get('label', '').strip()
+            if not label:
+                if image and not image_link:
+                    result['success'] = False
+                    result['errors'].append(
+                        "{0} has no text or img. Please make sure all {1}s "
+                        "have one or the other, or both.".format(noun, noun.lower()))
+                elif not image:
+                    result['success'] = False
+                    # If there's a bug in the code or the user just forgot to relabel a question,
+                    # votes could be accidentally lost if we assume the omission was an
+                    # intended deletion.
+                    result['errors'].append("{0} was added with no label. "
+                                            "All {1}s must have labels. Please check the form. "
+                                            "Check the form and explicitly delete {1}s "
+                                            "if not needed.".format(noun, noun.lower()))
+            if image:
+                # Labels might have prefixed space for markdown, though it's unlikely.
+                items.append((key, {'label': label, 'img': image_link.strip()}))
+            else:
+                items.append([key, label])
+
+        if not items:
+            result['errors'].append(
+                "You must include at least one {0}.".format(noun.lower()))
+            result['success'] = False
+
+        return items
+
+    def can_vote(self):
+        """
+        Checks to see if the user is permitted to vote. This may not be the case if they used up their max_submissions.
+        """
+        if self.max_submissions == 0:
+            return True
+        if self.max_submissions > self.submissions_count:
```

Comment:
@sarina `max_submissions == 0` means that there are no limit for resubmitting. Hence, `True` should be returned if either is true:

* max submission is zero
* `max_submisions` is **greater** than `submissions_count`.